share/availability/light/availability.go

```diff
@@ -130,7 +126,10 @@ func (la *ShareAvailability) SharesAvailable(ctx context.Context, header *header
 	}
 
 	// store the result of the sampling session
-	bs := encodeSamples(failedSamples)
+	bs, err := json.Marshal(failedSamples)
```

Comment:
Why do we break the serialization? We save a few loc for a much slower and bloated serialization. I don't think that's worth it at all.

Reply:
json encoding speed is 100-1000mb/s. It would take for our ~100 bytes encoding around 1ns. Does not seems to be a performance hit to me.

Reply:
The point is, there is no good reason to do it in the first place. Also, we don't need the JSON bloat.

Reply:
As you pointed out in original comment, the benefit is simplified code. Custom encoding does not bring much of performance benefit here.

Reply:
It does being storage overhead tho

Reply:
Discussed in sync that breaking is fine:
* We don't store successful samples, so the only way to prune those is by resyncing, which is equivalent to breaking. 
* We are still allowed to break

Reply:
Another missed thing is that the store state is changing in https://github.com/celestiaorg/celestia-node/pull/3887 to contain successful samples. We could store two states separately to not break, but again, it still requires resyncing to cleanup old samples.